IHP/ModelSupport.hs

```diff
@@ -608,12 +603,29 @@ deleteRecordById id = do
 -- >>> deleteRecords projects
 -- DELETE FROM projects WHERE id IN (..)
 deleteRecords :: forall record id. (?modelContext :: ModelContext, Show id, Table record, HasField "id" record id, ToField id) => [record] -> IO ()
-deleteRecords records = do
+deleteRecords records =
+    deleteRecordByIds (ids records)
+
+    -- do
+    -- let theQuery = "DELETE FROM " <> tableName @record <> " WHERE id IN ?"
+    -- let theParameters = PG.Only (PG.In (ids records))
+    -- sqlExec (PG.Query . cs $! theQuery) theParameters
+    -- pure ()
+{-# INLINABLE deleteRecords #-}
+
+-- | Like 'deleteRecordById' but for a list of Ids.
+--
+-- >>> let projectIds :: [ Id Project ] = ...
+-- >>> delete projectIds
+-- DELETE FROM projects WHERE id IN ('..')
+--
+deleteRecordByIds :: forall record id. (?modelContext :: ModelContext, Show id, Table record, HasField "id" record id, ToField id) => [id] -> IO ()
```

Comment:
The mentioned type error is caused by the unused type variable `record` here :) As there's only the ids and no record passed to this function, GHC cannot figure out the type here.

Removing all mentions of `record` in the type signature will fix the issue:
```suggestion
deleteRecordByIds :: forall id. (?modelContext :: ModelContext, Show id, ToField id) => [id] -> IO ()
```

Reply:
It has `tableName @record` in the function, so I think it's still needed

Reply:
Good point 👍  missed that